src/main/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtil.java

```diff
@@ -26,22 +27,33 @@
 public class AutoSetFileLinksUtil {
 
     private static final Log LOGGER = LogFactory.getLog(AutoSetLinks.class);
+    private List<Path> directories;
+    private AutoLinkPreferences autoLinkPreferences;
+    private ExternalFileTypes externalFileTypes;
 
-    public List<LinkedFile> findassociatedNotLinkedFiles(BibEntry entry, BibDatabaseContext databaseContext, FileDirectoryPreferences fileDirPrefs, AutoLinkPreferences autoLinkPrefs, ExternalFileTypes externalFileTypes) {
+    public AutoSetFileLinksUtil(BibDatabaseContext databaseContext, FileDirectoryPreferences fileDirectoryPreferences, AutoLinkPreferences autoLinkPreferences, ExternalFileTypes externalFileTypes) {
+        this(databaseContext.getFileDirectoriesAsPaths(fileDirectoryPreferences), autoLinkPreferences, externalFileTypes);
+    }
+
+    public AutoSetFileLinksUtil(List<Path> directories, AutoLinkPreferences autoLinkPreferences, ExternalFileTypes externalFileTypes) {
```

Comment:
I am against creating a second constructor. It was my initial idea in #3368  and guess who supposed the current alternative? 😆 
See your comment https://github.com/JabRef/jabref/pull/3368#discussion_r147604206

Reply:
I don't understand what you mean. The linked comment is about the list of entries and not about how the `AutoSetFileLinksUtil` gets the things needed (like directories and preferences). I think, the current setup makes sense: on initialization, specify everything the auto set functionality needs to know. Why don't you like these constructors?

Reply:
Ah sorry, I could not see all code in the line on my mobile, so I was a b9it confused. Forget about my remark,

Reply:
No problem!